docs/src/examples/components/Popup/Variations/PopupExamplePosition.shorthand.tsx

```diff
@@ -1,145 +1,67 @@
 import React from 'react'
 import { Button, Grid, Popup } from '@stardust-ui/react'
 
+class PopupArrowExample extends React.Component<any, any> {
+  state = { popupOpen: false }
+
+  togglePopupState = () => {
+    this.setState(prev => ({ popupOpen: !prev.popupOpen }))
+  }
+
+  render = () => {
+    const { position, align, icon, padding } = this.props
+
+    const buttonStyles = { padding, height: '38px', minWidth: '64px' }
+
+    return (
+      <Popup
+        open={this.state.popupOpen}
+        align={align}
+        position={position}
+        trigger={
+          <Button
+            onClick={() => this.togglePopupState()}
```

Comment:
we redefine the function on every click, bad for perf and not good react practice; since `togglePopupState` is already an arrow function u can do
```typescript
onClick={this. togglePopupState}
```

Reply:
have already adressed that in comment, would leave it as it is as it goes along with official React docs